src/Carousel/index.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { FunctionComponent, ReactNode, cloneElement, isValidElement, useRef } from 'react'
+
+import carouselHelper from './carouselHelper'
+import clsx from 'clsx'
+import styles from './carousel.module.scss'
+
+interface Props {
+  /** Aria label to set apart different carousels in the page.  
+   * Label is prefixed with 'carousel-'
+   */
+  ariaLabel: string
+  /** Sets the class for the Carousel wrapper */
+  carouselClass?: string
+  /** Sets how many slides to show */
+  itemsToShow?: number
+  /** Sets the transition control button alignments. Two non conflicting configurations can be combined.  
+   * Valid configurations are: 'top', 'middle', 'center', 'apart', 'left', 'right', 'bottom'.  
+   * 'middle' centers vertically while 'center' centers horizontally. */
+  controlAlignment?: string 
+  /** Hides control buttons unless hovered */
+  hideControls?: boolean
+  /** Sets the indicator buttons' style */
+  indicatorStyle?: 'bar' | 'round'
+  /** Hides indicator buttons */
+  hideIndicators?: boolean
+  /** Supply a px margin between slides */
+  layoutGap?: number
+  /** Allows Carousel to be cyclical. */
+  infinite?: boolean
+  /** Enables automatic transitions. */
+  autoSlide?: boolean
+  /** Time in milliseconds for autoSlide */
+  duration?: number
+  [rest: string]: unknown // ...rest property
+}
+
+const Carousel: FunctionComponent<Props> = ({
+  carouselClass,
+  ariaLabel,
+  itemsToShow = 1,
+  controlAlignment = 'middle apart',
+  hideIndicators = false,
+  hideControls = false,
+  indicatorStyle = 'round',
+  layoutGap = 0,
+  infinite = false,
+  autoSlide = false,
+  duration = 3000,
+  children
+}) => {
+  const {
+    slidesRef,
+    slidesLeft,
+    slides,
+    slidesWidth,
+    indicators,
+    indicatorRef,
+    alignment,
+    slidesTransition,
+    ariaLive,
+    handleDragStart,
+    handleDragEndHandler,
+    handleDragActionHandler,
+    handleIndexCheck,
+    shiftSlide
+  } = carouselHelper(
+    children,
+    itemsToShow,
+    controlAlignment,
+    hideIndicators,
+    indicatorStyle,
+    duration,
+    infinite,
+    autoSlide,
+    layoutGap
+  )
+
+  const buttons = (
+    <>
+      <button
+        id="prev"
+        aria-hidden={(hideControls && 'true') || 'false'}
+        className={clsx(
+          styles['carousel-wrapper-control'],
+          styles['prev'],
+          hideControls && styles['hidden'],
+          alignment
+        )}
+        onClick={() => shiftSlide(-1)}
+      >
+        <p className={clsx(styles['sr-only'])}>Move Slider Left Button.</p>
+      </button>
+      <button
+        id="next"
```

Comment:
I would make this more unique somehow. There should only be 1 "next" id on a page at a time. What if there's multiple carousels on the page?